### PR TITLE
Enable The Optional Sanitizer (by default)

### PR DESCRIPTION
**Pull request by CodaFi**
Suffer no longer under the silent tyranny of the force-unwrap!

The Optional Sanitizer is a revolutionary new tool that allows
developers to gain enhanced insight into the inner workings of their
programs.  Traditional debugging techniques rely on a boring terminal
with text scrolling by, and a thousand commands that you need to
remember.  Instead, we have simplified the process of debugging down
to, what we see as, the most unused sense when programming: The ears.

That’s right, now you can hear your program as it executes! No longer
will you have to struggle with the debugger, now every optional
force-unwrap can grate on your senses with the ease and simplicity that
only Apple can provide.

Tooling is also highly accessible.  To those that wish to get all the
benefits of OSan, but may not be able to experience the aural sensation to its
fullest extent, we provide a visual OSan that is enabled by passing
`-Xfrontend -visual-osan-unwrap-instrumentation`

*A future patch will enable the ability to disable OSan.

### CODE REVIEW

**Comment by CodaFi** [comment]
`<unfools>` (For the love of god, don't trigger CI) `</unfools>`

**Comment by dabrahams** [comment]
+1, but the standard alert sound is insufficiently annoying. Please provide a custom nails-on-chalkboard sound

**Comment by CodaFi** [comment]
@gparker42 Would it be alright if I embedded the nails-on-chalkboard m4a into the runtime and shell out to iTunes?

**Comment by harlanhaskins** [comment]
[libc shells out to perl](https://github.com/Apple-FOSS-Mirror/Libc/blob/2ca2ae74647714acfc18674c3114b1a5d3325d7d/gen/wordexp.c#L192), so I think that's fair.

**Comment by zisko** [comment]
I'm keeping this patch around in case @shahmishal gets locked in the lab

**Comment by devincoughlin** [comment]
This is awesome. Made my day.

File: stdlib/public/runtime/Errors.cpp [hunk 1]
```diff
@@ -288,6 +296,43 @@ swift_deletedMethodError() {
...
 
+static void *Swift_Framework_Token = NULL;
+static void *Swift_OSan_Impl;
+
+SWIFT_RUNTIME_EXPORT
+void
+swift_osan_unwrap(bool aural) {
+  if (Swift_Framework_Token == NULL) {
+    if (aural) {
+      Swift_Framework_Token = dlopen("/System/Library/Frameworks/AudioToolbox.framework/Versions/A/AudioToolbox", RTLD_FIRST);
```
Comment (hughbe) [inline stdlib/public/runtime/Errors.cpp]: I'd love to bring such an important feature to Windows. This is too platform specific :'(

**Review by hughbe — CHANGES_REQUESTED** [review]

**Pull request closed without merging**